tests/remote-environment-controller-tests/test/testkit/k8sresources_check.go

```diff
@@ -0,0 +1,128 @@
+}
+
+func (checker *k8sChecker) checkDeployments(t *testing.T, shouldFail bool, errCheckFunc func(*testing.T, error), resourceCheckFunc func(*testing.T, interface{})) {
+	gatewayName := fmt.Sprintf(gatewayNameFormat, checker.remoteEnvName)
+	eventServiceName := fmt.Sprintf(eventServiceNameFormat, checker.remoteEnvName)
+
+	gatewayDeploy, err := checker.getDeploymentWithRetries(gatewayName, v1.GetOptions{}, shouldFail)
+	errCheckFunc(t, err)
+	resourceCheckFunc(t, gatewayDeploy)
+
+	eventsDeploy, err := checker.client.GetDeployment(eventServiceName, v1.GetOptions{})
+	errCheckFunc(t, err)
+	resourceCheckFunc(t, eventsDeploy)
+}
+
+func (checker *k8sChecker) checkIngress(t *testing.T, errCheckFunc func(*testing.T, error), resourceCheckFunc func(*testing.T, interface{})) {
+	ingressName := fmt.Sprintf(gatewayNameFormat, checker.remoteEnvName)
+
+	ingress, err := checker.client.GetIngress(ingressName, v1.GetOptions{})
+	errCheckFunc(t, err)
+	resourceCheckFunc(t, ingress)
+}
+
+func (checker *k8sChecker) checkRole(t *testing.T, errCheckFunc func(*testing.T, error), resourceCheckFunc func(*testing.T, interface{})) {
+	roleName := fmt.Sprintf(gatewayRoleFormat, checker.remoteEnvName)
+
+	role, err := checker.client.GetRole(roleName, v1.GetOptions{})
+	errCheckFunc(t, err)
+	resourceCheckFunc(t, role)
+}
+
+func (checker *k8sChecker) checkRoleBinding(t *testing.T, errCheckFunc func(*testing.T, error), resourceCheckFunc func(*testing.T, interface{})) {
+	roleBindingName := fmt.Sprintf(gatewayRoleBindingFormat, checker.remoteEnvName)
+
+	role, err := checker.client.GetRoleBinding(roleBindingName, v1.GetOptions{})
+	errCheckFunc(t, err)
+	resourceCheckFunc(t, role)
+}
+
+func (checker *k8sChecker) checkServices(t *testing.T, errCheckFunc func(*testing.T, error), resourceCheckFunc func(*testing.T, interface{})) {
+	gatewayApiName := fmt.Sprintf(gatewayApiFormat, checker.remoteEnvName)
+	gatewayEchoName := fmt.Sprintf(gatewayEchoFormat, checker.remoteEnvName)
+	eventsApiName := fmt.Sprintf(eventServiceApiFormat, checker.remoteEnvName)
+
+	gatewayApiSvc, err := checker.client.GetService(gatewayApiName, v1.GetOptions{})
+	errCheckFunc(t, err)
+	resourceCheckFunc(t, gatewayApiSvc)
+
+	gatewayEchoSvc, err := checker.client.GetService(gatewayEchoName, v1.GetOptions{})
+	errCheckFunc(t, err)
+	resourceCheckFunc(t, gatewayEchoSvc)
+
+	eventsSvc, err := checker.client.GetService(eventsApiName, v1.GetOptions{})
+	errCheckFunc(t, err)
+	resourceCheckFunc(t, eventsSvc)
+}
+
+func (checker *k8sChecker) getDeploymentWithRetries(name string, options v1.GetOptions, shouldFail bool) (*v1apps.Deployment, error) {
+	var deployment *v1apps.Deployment
+	var err error
+
+	for i := 0; i < checker.retryCount; i++ {
+		deployment, err = checker.client.GetDeployment(name, v1.GetOptions{})
+		if !shouldRetry(err, shouldFail) {
+			break
+		}
+		time.Sleep(checker.retryWaitTime)
+	}
+
+	return deployment, err
+}
+
+func shouldRetry(err error, shouldFail bool) bool {
```

Comment:
This function is not easy to read since it covers two scenarios : checks for adding and removal.Please, get rid of it.